rotkehlchen/chain/bitcoin/bch/manager.py

```diff
@@ -0,0 +1,101 @@
+from rotkehlchen.utils.misc import satoshis_to_btc
+from rotkehlchen.utils.network import request_get, request_get_dict
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BitcoinCashManager(BitcoinCommonManager):
+
+    def __init__(self, database: 'DBHandler') -> None:
+        haskoin_api = BtcApiCallback(name='haskoin', function_name='_query_haskoin_{0}')
+        # TODO: add other APIs for fallbacks if haskoin goes down
+        super().__init__(
+            database=database,
+            blockchain=SupportedBlockchain.BITCOIN_CASH,
+            api_callbacks={
+                BtcQueryAction.BALANCES: [haskoin_api],
+                BtcQueryAction.HAS_TRANSACTIONS: [haskoin_api],
+                # TODO: add support for querying transactions
+            },
+        )
+
+    @staticmethod
+    def _get_haskoin_address(
+            address_from_api: str,
+            requested_addresses: Sequence[BTCAddress],
+    ) -> BTCAddress | None:
+        """Get an address from a haskoin API response.
+        This is needed because the API only returns CashAddr format.
+        """
+        address: BTCAddress | None
+        if (
+            (address := BTCAddress(address_from_api)) in requested_addresses or
+            (address := BTCAddress(address_from_api.split(':')[1])) in requested_addresses or
+            (address := cash_to_legacy_address(address_from_api)) is not None
+        ):
+            return address
+
+        return None
+
+    def _query_haskoin_balances(
+            self,
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, FVal]:
+        """Query a mapping of addresses to balances from haskoin."""
+        balances: dict[BTCAddress, FVal] = {}
+        accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
+        for accounts_chunk in accounts_chunks:
+            params = ','.join(accounts_chunk)
+            bch_resp = request_get(url=f'{HASKOIN_BASE_URL}/bch/address/balances?addresses={params}')  # noqa: E501
+            for entry in bch_resp:
+                if (address := self._get_haskoin_address(
+                        address_from_api=entry['address'],
+                        requested_addresses=accounts_chunk,
+                )) is not None:
+                    balances[address] = satoshis_to_btc(deserialize_fval(
+                        value=entry['confirmed'],
+                        name='balance',
+                        location='bitcoin cash balance querying',
+                    ))
+
+        return balances
+
+    def _query_haskoin_has_transactions(
+            self,
+            accounts: Sequence[BTCAddress],
+    ) -> dict[BTCAddress, tuple[bool, FVal]]:
+        """Query a mapping of which addresses have had transactions
+        and also their current balances from haskoin.
```

Comment:
same about the errors. Why do you need if it has transactions instead of just querying it?

Reply:
ah, is this to detect if an address is active?